### PR TITLE
Fix MQTT transport deprecations

### PR DESCRIPTION
**Pull request by wborn**
Related to https://github.com/openhab/openhab-core/pull/1671

### CODE REVIEW

**Review by kaikreuzer — APPROVED** [review]

File: bundles/org.openhab.binding.mqtt/src/main/java/org/openhab/binding/mqtt/handler/BrokerHandler.java [hunk 1]
```diff
@@ -223,8 +223,6 @@ protected MqttBrokerConnection createBrokerConnection() throws IllegalArgumentEx
             connection.setTimeoutExecutor(scheduler, TIMEOUT_DEFAULT);
         }
 
-        connection.setRetain(config.retainMessages);
```
Comment (cpmeister) [inline bundles/org.openhab.binding.mqtt/src/main/java/org/openhab/binding/mqtt/handler/BrokerHandler.java:226]: Should the config also be removed or is it used elsewhere?

Comment (wborn) [reply bundles/org.openhab.binding.mqtt/src/main/java/org/openhab/binding/mqtt/handler/BrokerHandler.java:226]: It's still used in other locations of this file.

Comment (cpmeister) [reply bundles/org.openhab.binding.mqtt/src/main/java/org/openhab/binding/mqtt/handler/BrokerHandler.java:226]: oh ok

**Review by cpmeister — APPROVED** [review]

**Comment by kaikreuzer** [comment]
@wborn Might you have missed something here? https://ci.openhab.org/job/openHAB-Addons/61/console

**Pull request merged**